packages/ui/src/machines/authenticator/actors/signIn.ts

```diff
@@ -549,6 +556,20 @@ export function signInActor({ services }: SignInMachineOptions) {
         shouldSetupTOTP: (_, event): boolean => {
           return isExpectedChallengeName(getChallengeName(event), 'MFA_SETUP');
         },
+        /* START ISSUE 3767 - Temporary fix. Will change to handle 'SELECT_MFA_TYPE' */
+        isSelectMFATypeChallenge: (_, event): boolean => {
+          const isSelectMfaTypeChallenge = isExpectedChallengeName(
```

Comment:
I prefer to change it back to `shouldSelectMFAType` to make it consistent with `shouldSetupTOTP` etc. . Since @wlee221 commented on this earlier, what's your thought?

Reply:
I don't think we should use the word `should`, since that says Authenticator "should" do something, which we do not. I'm okay with giving slight modifications to the second variable name to have them different, or come up with some better name that starts with `should` (e.g. `shouldWarnSelectMFAChallenge`, but shorter)...

Happy to sync offline too.

Reply:
Done 0d9d07f19f1ddd26804daa5e6865a4845fb58bd3